add parameter to blur background image

Nikfinn99 opened the pull request:
fixes #1 
example config:

```
media_art_background:
  blur: 10px
  entities:
    - entity: media_player.spotify
      valid_states:
        - playing
      image_attribute: entity_picture
```

Nikfinn99 committed:
add parameter to blur background image

example config:

media_art_background:
  blur: 10px
  entities:
    - entity: media_player.spotify
      valid_states:
        - playing
      image_attribute: entity_picture

TheLastProject commented on media-art-background.js:
```diff
@@ -1,9 +1,10 @@
     let entityName = entity.entity || entity;
     let entityValidStates = entity.valid_states || ['playing'];
     let entityImageAttribute = entity.image_attribute || 'entity_picture';
+    let filterBlur = lovelace.config.media_art_background.blur || '0px';
```
This is a "global" config setting, so please read it before going into the loop, as it's not entity-specific.

TheLastProject commented on media-art-background.js:
```diff
@@ -17,23 +18,37 @@ function setBackground(root, lovelace, app, originalStyle) {
 
-setInterval(function() { setBackground(root, lovelace, app, originalStyle) }, 5000);
-setBackground(root, lovelace, app, originalStyle);
+const bground = document.createElement("div"); // create empty container for background 
```
Please use camelCase naming like the rest of the variables

TheLastProject commented on media-art-background.js:
```diff
@@ -1,9 +1,10 @@
-function setBackground(root, lovelace, app, originalStyle) {
+function setBackground(root, lovelace, bground_elem, originalStyle) {
```
Please use camelCase naming like the rest of the variables

TheLastProject commented on media-art-background.js:
```diff
@@ -17,23 +18,37 @@ function setBackground(root, lovelace, app, originalStyle) {
+
+app.appendChild(bground);
+
+const originalStyle = bground.style;
```
Hmm, not sure if this style resetting is still the best way because of adding a div now. But should still work fine so sure :)

TheLastProject commented:
Oh yeah, and this'll need to be documented in the README too :)

Nikfinn99 committed:
Update Documentation for new blur feature

Nikfinn99 committed:
correct camelCase variable names, refactor code

- used camelCase for variable names
- refactor applying of styles
-> storing in originalStyle did not work properly
-> new setupStyle method now also processes global config variables

Nikfinn99 committed:
Merge branch 'master' into patch-2

Nikfinn99 committed:
Update Documentation for new transition feature

TheLastProject commented on media-art-background.js:
```diff
@@ -17,23 +46,27 @@ function setBackground(root, lovelace, app, originalStyle) {
   }
 
-  app.style = originalStyle;
+  setupStyle(lovelace, bgroundElem);
```
I'm pretty sure this no longer unsets the media background now

Nikfinn99 replied:
yes but by setting opacity to 0 the image will be hidden and if the new transition parameter is configured, it will also fade in or out the background

Nikfinn99 replied:
so basically the image will still be there but it is transparent

TheLastProject replied:
Oh, I totally didn't notice that while reading the code, fancy! Sadly whenever the entity picture changes it is still instant, but I totally love the rest of this. Kinda feel that a 10px blur and 5s transition should be the default after playing with your version a bit

Nikfinn99 replied:
yes maybe i will add a crossfade between backgrounds in the soon :)
and i agree that for most users different defaults would be better to really have a nice background

TheLastProject replied:
Well, if you want, I can already merge this because it's definitely an improvement? :)

Nikfinn99 replied:
yes you could :)
i will add additional features in a separate pr
